pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -9,14 +10,22 @@ import (
 
 	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 
 	commonapiv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/common/v1beta1"
 	experimentsv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/experiments/v1beta1"
+	"github.com/kubeflow/katib/pkg/controller.v1beta1/consts"
 	"github.com/kubeflow/katib/pkg/controller.v1beta1/experiment/manifest"
+	util "github.com/kubeflow/katib/pkg/controller.v1beta1/util"
 	jobv1beta1 "github.com/kubeflow/katib/pkg/job/v1beta1"
 	mccommon "github.com/kubeflow/katib/pkg/metricscollector/v1beta1/common"
+
+	pytorchv1 "github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1"
+	tfv1 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1"
```

Comment:
Were we doing this validation before? Can we avoid operator imports somehow?

Reply:
Before that we didn't validate that Job was added properly: https://github.com/kubeflow/katib/blob/master/pkg/webhook/v1alpha3/experiment/validator/validator.go#L161-L170. We just checked Kind.

Why do we want to avoid these imports?
That can help user to submit incorrect TFJobs, Jobs or PyTorch Jobs.

Reply:
The general idea is to decouple katib from operators(if we can)

Reply:
In the future yes.
After we implement general provider for all K8s CRDs we can remove this validation.
I think for now we can keep this validation to make sure that current Trial jobs are correct